solutions/bio1/Parking/index.tsx

```diff
@@ -0,0 +1,49 @@
+import { SCodeBlock, SCode, SText, STitle } from "components";
+
+export const Parking = () => {
+  return (
+    <>
+      <STitle>Idea</STitle>
+      <SText>
+        This type of problem (find $k-$th of something) is quite common as problem 3, and the solution to this one can be readily applied to others of its type.
+        Clearly, constructing all possible preference lists, then sorting them to find the ith one is much too slow, so you need to find a way to construct the solution letter by letter. 
+        You do this by iterating through each car (a to z), and figuring out which preference spot to give each car so that the final list is the $k-th$ lexographically, while still maintaining a valid preference list.
+      </SText>
+
+      <STitle>Solution</STitle>
+      <SText>
+        Taking the sample as an example, it is clear the parking slots progress like this:
+      </SText>
+      <SText>
+      <SCode>.... -{'>'} .a... -{'>'} .ab. -{'>'} cab. -{'>'} cabd</SCode>
+      </SText>
+
+      <SText>
+        So at each stage, we want to find where we can place the next car. For it to go in its correct slot, it must prefer the slot either that it ends up in, or a slot in front of that one which has no empty spaces between it and the slot which the car ends up in.
+        For example, from cab. -{'>'} cabd, d can have any preferred parking slot, as all the slots in front of it are filled, and the only empty slot is D, which is a valid preferred slot as that is the one the car d ends up in. 
+        On the other hand, if we have an arrangement like a.bc., and the next arrangement as a.bcd, D can only go in slots C, D, and E. Even though slot A is filled, there is an empty slot in between it and the E slot which car d ends up in, so it cannot be d's preferred slot. 
```

Comment:
I won't tell you every single mistake you've made, but all the ABCD's should have latex on them